wren-ui/src/components/pages/home/promptThread/CollapseContent.tsx

```diff
@@ -32,14 +55,53 @@ export default function CollapseContent(props: Props) {
     sql,
     previewDataResult,
     attributes,
+    onChangeNativeSQL,
+    nativeSQLResult,
   } = props;
 
+  const { hasNativeSQL, dataSourceType } = nativeSQLResult;
+  const showNativeSQL = Boolean(attributes.isLastStep) && hasNativeSQL;
+
   return (
     <>
       {(isViewSQL || isViewFullSQL) && (
-        <pre className="p-0 my-3">
-          <CodeBlock code={sql} showLineNumbers maxHeight="300" />
-        </pre>
+        <StyledPre className="p-0 my-3" showNativeSQL={showNativeSQL}>
+          {showNativeSQL && (
+            <StyledToolBar className="d-flex justify-space-between">
+              <div>
+                <Image
+                  className="mr-2"
+                  src={DATA_SOURCE_OPTIONS[dataSourceType].logo}
+                  alt={DATA_SOURCE_OPTIONS[dataSourceType].label}
+                  width="22"
+                  height="22"
+                />
+                <Text className="gray-8 text-medium text-sm">
+                  {DATA_SOURCE_OPTIONS[dataSourceType].label}
+                </Text>
+              </div>
+              <div>
+                <Switch
+                  checkedChildren={<CheckOutlined />}
+                  unCheckedChildren={<CloseOutlined />}
+                  className="mr-2"
+                  size="small"
+                  onChange={onChangeNativeSQL}
+                  loading={nativeSQLResult.loading}
+                />
+                <Text className="gray-8 text-medium text-sm">
+                  Show original SQL
+                </Text>
+              </div>
+            </StyledToolBar>
+          )}
+          <CodeBlock
```

Comment:
(NIT) For UX, there is flicking when switching the native SQL toggle, I think it is because of the loading height.
Also, the information bar font type has been changed since it's wrapped in a `pre` element.

https://github.com/Canner/WrenAI/assets/9657305/359360a2-b6e9-4a8e-a99a-74dd62d52786

Reply:
Thanks for discussing it with me, fixed.